backend-server/src/main/java/org/dice/ida/vizsuggest/BarGraph.java

```diff
@@ -0,0 +1,57 @@
+package org.dice.ida.vizsuggest;
+
+import org.dice.ida.model.AttributeSummary;
+import org.dice.ida.model.DataSummary;
+import org.dice.ida.model.VisualizationSuggestion;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A class to suggest bar graph parameters to the users for the given dataset
+ * @author Nandeesh & Sourabh
+ */
+public class BarGraph implements IVisualizationParent{
+
+    /**
+     * Function to decide best suitable attributes for bar graph in the given dataset
+     * @param dataSummary  data summary object of the dataset
+     * @return Object of visualization suggestion class having the list of x-axis and corresponding y-axis parameters
+     */
+    @Override
+    public VisualizationSuggestion getParams(DataSummary dataSummary) {
+        VisualizationSuggestion visualizationSuggestion = new VisualizationSuggestion();
+        List<AttributeSummary> xAxis = dataSummary.getAttributeSummaryList()
+                .stream()
+                .filter(x -> "Nom".equals(x.getType()) && x.getDiscreteValuesCount() <= 50).collect(Collectors.toList());
```

Comment:
Numerical constants like this (and the following ones) should be assigned to a constant variable